eng/Versions.props

```diff
@@ -42,7 +42,7 @@
       TODO: Remove pinned compiler version once arcade supplies runtime with a compiler capable of handling !!
       and has https://github.com/dotnet/roslyn/pull/59776
     -->
-    <MicrosoftNetCompilersToolsetVersion>4.2.0-2.22128.1</MicrosoftNetCompilersToolsetVersion>
+    <MicrosoftNetCompilersToolsetVersion>4.3.0-1.22201.1</MicrosoftNetCompilersToolsetVersion>
```

Comment:
Given this is `main`/`main-vs-deps` and therefore not part of the `release/dev17.2` branch, is this expected to cause any issues in VS or other tooling by us pre-emptively moving onto it?

If not, then `user defined checked operators` was merged into Roslyn's `main-vs-deps` branch and is also `4.3.0-*` (will be merged into `main` with `https://github.com/dotnet/roslyn/pull/60515`), so it would be great to pull that in at the same time if feasible.

Reply:
> so it would be great to pull that in at the same time if feasible

We can easily update again to pull it in.

Right now we need the compiler crash addressed first, though.

Reply:
Just noting that `4.3.0-1.22206.2` or later (at the very least, didn't check prior builds) contains `user-defined checked operators`

Reply:
Also noting that `4.3.0-1.2206.2` has a different set of errors:
```
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\RSA\RSAKeyPemTests.cs(403,21): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyFileTests.cs(71,62): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyPemTests.cs(268,21): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyPemTests.cs(349,25): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyPemTests.cs(368,25): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\RSA\RSAKeyFileTests.cs(72,62): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyFileTests.cs(584,31): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\RSA\RSAKeyFileTests.cs(1291,31): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Csp\tests\System.Security.Cryptography.Csp.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\PrivateKeyHelpers.cs(22,29): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, PbeParameters)' and 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, PbeParameters)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\PrivateKeyHelpers.cs(50,29): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, PbeParameters)' and 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, PbeParameters)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\PrivateKeyHelpers.cs(78,31): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, PbeParameters)' and 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, PbeParameters)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(22,65): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<byte>, ReadOnlyMemory<byte>, out int)' and 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<char>, ReadOnlyMemory<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs12\ShroudedKeyBagTests.cs(33,26): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(226,43): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, PbeParameters)' and 'AsymmetricAlgorithm.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, PbeParameters)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(235,47): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<byte>, ReadOnlyMemory<byte>, out int)' and 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<char>, ReadOnlyMemory<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(243,47): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<byte>, ReadOnlyMemory<byte>, out int)' and 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<char>, ReadOnlyMemory<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(246,47): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<byte>, ReadOnlyMemory<byte>, out int)' and 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<char>, ReadOnlyMemory<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(248,70): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<byte>, ReadOnlyMemory<byte>, out int)' and 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<char>, ReadOnlyMemory<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(272,37): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, PbeParameters, Span<byte>, out int)' and 'AsymmetricAlgorithm.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, PbeParameters, Span<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(283,70): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<byte>, ReadOnlyMemory<byte>, out int)' and 'Pkcs8PrivateKeyInfo.DecryptAndDecode(ReadOnlySpan<char>, ReadOnlyMemory<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(354,65): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.Encrypt(ReadOnlySpan<byte>, PbeParameters)' and 'Pkcs8PrivateKeyInfo.Encrypt(ReadOnlySpan<char>, PbeParameters)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\System.Security.Cryptography.Pkcs\tests\Pkcs8PrivateKeyInfoTests.cs(358,28): error CS0121: The call is ambiguous between the following methods or properties: 'Pkcs8PrivateKeyInfo.TryEncrypt(ReadOnlySpan<byte>, PbeParameters, Span<byte>, out int)' and 'Pkcs8PrivateKeyInfo.TryEncrypt(ReadOnlySpan<char>, PbeParameters, Span<byte>, out int)' [.\src\libraries\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\EC\ECKeyFileTests.cs(96,62): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\EC\ECKeyPemTests.cs(325,21): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\EC\ECKeyPemTests.cs(403,25): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\EC\ECKeyPemTests.cs(432,25): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\RSA\RSAKeyPemTests.cs(403,21): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\EC\ECKeyFileTests.cs(983,31): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\EC\ECKeyFileTests.cs(1084,35): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyFileTests.cs(71,62): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\RSA\RSAKeyFileTests.cs(72,62): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyPemTests.cs(268,21): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyPemTests.cs(349,25): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyPemTests.cs(368,25): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<byte>)' and 'AsymmetricAlgorithm.ImportFromEncryptedPem(ReadOnlySpan<char>, ReadOnlySpan<char>)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\RSA\RSAKeyFileTests.cs(1291,31): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
.\src\libraries\Common\tests\System\Security\Cryptography\AlgorithmImplementations\DSA\DSAKeyFileTests.cs(584,31): error CS0121: The call is ambiguous between the following methods or properties: 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>, ReadOnlySpan<byte>, out int)' and 'AsymmetricAlgorithm.ImportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>, ReadOnlySpan<byte>, out int)' [.\src\libraries\System.Security.Cryptography\tests\System.Security.Cryptography.Tests.csproj]
```

One of the failures in question looks like:
```csharp
string pem = @"
-----BEGIN ENCRYPTED PRIVATE KEY-----
MIIBsTBLBgkqhkiG9w0BBQ0wPjApBgkqhkiG9w0BBQwwHAQIcvgI1lw9LqYCAggA
MAwGCCqGSIb3DQIJBQAwEQYFKw4DAgcECFDpLREQXt5pBIIBYOKuM5ljAvCViDL+
nTFq7A/fI9rqdL20TMdf0wy7s43oXmsw5gCStoNEaoVToFCQWYYBRU99mK8YNFA8
1ZJT53SDS7buJ0zX9oDltf2ByXRPI4mn2Il2HZvN2hi9ir1w8M3XoSFSurN9tC8r
IOiGkVfK9Ll54knONewNiCNefFZFctRfVMbac5SwHokCkBMHukl0oPrpVuBE8kRo
p7XtjM8ILtzLVz0iLqKXiNIf6kRdouCBmCn8VIQgIvPPIHD8vheMXWjN7g69P5n4
1YI4c/acljcofmq1BBPTwvxaETrg2NHW0XMIgAxoaVP8lIIGlNk1glWTYpuMd69L
AWvBUt33Sozc+dF0l7NGLAWL2tqkkpyDQuKn6UgYz/vxkFeQAVfSuaJVR+fUlHg0
N4lD7/hJq7b+yYPhlN3Fvvt8M9MtRg1TLAve67CA2v4TITHB06M/ELe3y42bZuLW
CA7ffFk=
-----END ENCRYPTED PRIVATE KEY-----";
                rsa.ImportFromEncryptedPem(pem, "test");
```

Reply:
Logged https://github.com/dotnet/roslyn/issues/60612